02_Development/Display/sfm3300d.py

```diff
@@ -0,0 +1,112 @@
+import time
+import math
+
+import crcmod
+
+from i2c_interface import I2CInterface
+
+
+class FlowSensor():
+    """A class!!
+    Main Datasheet:
+    https://www.mouser.com/datasheet/2/682/Sensirion_Mass_Flow_Meters_SFM3300_Datasheet-1524535.pdf
+    """
+
+    def __init__(self):
+        self._calibrator = Calibrator(SensorConstants.OFFSET_FLOW,
+                                      SensorConstants.SCALE_FACTOR_FLOW)
+        self._communicator = Communicator()
+
+    def close(self):
+        self._communicator.close()
+
+    def init_flow(self):
+        self._communicator.init_flow()
+        self._communicator.raw_flow()  # the first datum is garbage
+
+    def flow(self):
+        return self._calibrator.flow(self._communicator.raw_flow())
+
+    def serial_number(self):
+        return self._communicator.serial_number()
+
+
+class Calibrator():
+
+    def __init__(self, offset_flow, scale_factor_flow):
+        if math.isclose(scale_factor_flow, 0.0):
+            raise ZeroDivisionError("scale_factor_flow cannot be zero.")
```

Comment:
What happens to the program if this exception gets raised?  Does the entire device crash and we have to hard restart?

Reply:
Yeah the program would probably crash since `scale_factor_flow` is a constant that you need to have right for calibration.  It shouldn't ever be 0 though since that constant is defined to be 120 or so in that module.